harness/determined/common/context.py

```diff
@@ -1,223 +1,173 @@
-            root_path, sizeof_fmt(0), 0
-        )
-        print(msg, end="\r", flush=True)
-
-        # We could use pathlib.Path.rglob for scanning the directory;
-        # however, the Python documentation claims a warning that rglob may be
-        # inefficient on large directory trees, so we use the older os.walk().
-        for parent, dirs, files in os.walk(str(root_path)):
-            for directory in dirs:
-                dir_path = pathlib.Path(parent).joinpath(directory)
-                dir_rel_path = dir_path.relative_to(root_path)
-
-                # If the file matches any path specified in .detignore, then ignore it.
-                if ignore_spec.match_file(str(dir_rel_path) + "/"):
-                    continue
-
-                # Determined only supports POSIX-style file paths.  Use as_posix() in case this code
-                # is executed in a non-POSIX environment.
-                entry_path = dir_rel_path.as_posix()
-
-                context.add_item(ContextItem.from_local_dir(entry_path, dir_path))
-
-            for file in files:
-                file_path = pathlib.Path(parent).joinpath(file)
-                file_rel_path = file_path.relative_to(root_path)
-
-                # If the file is the .detignore file or matches one of the
-                # paths specified in .detignore, then ignore it.
-                if file_rel_path.name == ".detignore":
-                    continue
-                if ignore_spec.match_file(str(file_rel_path)):
-                    continue
-
-                # Determined only supports POSIX-style file paths.  Use as_posix() in case this code
-                # is executed in a non-POSIX environment.
-                entry_path = file_rel_path.as_posix()
-
-                try:
-                    entry = ContextItem.from_local_file(entry_path, file_path)
-                except OSError:
-                    print("Error reading '{}', skipping this file.".format(entry_path))
-                    continue
-
-                context.add_item(entry)
-                if context.size > limit:
-                    print()
-                    raise ValueError(
-                        "Directory '{}' exceeds the maximum allowed size {}.\n"
-                        "Consider using a .detignore file to specify that certain files "
-                        "or directories should be omitted from the model.".format(
-                            root_path, sizeof_fmt(constants.MAX_CONTEXT_SIZE)
-                        )
+    items = []
+    size = 0
+
+    def add_item(f: bindings.v1File) -> None:
+        nonlocal size
+        items.append(f)
+        size += v1File_size(f)
+
+    local_path = local_path.resolve()
+
+    if not local_path.exists():
+        raise Exception("Path '{}' doesn't exist".format(local_path))
+
+    if local_path.is_file():
+        raise ValueError("Path '{}' must be a directory".format(local_path))
```

Comment:
f-string?

Reply:
sure.  I just indented this code and didn't see it, but I'm happy to fix them.